lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -410,6 +473,11 @@ def main():
                 except IOError as e:
                     module.fail_json(msg=str(e))
 
+        # Tag Function
+        if tags is not None and len(tags) >= 0:
+            if set_tag(client, module, tags, current_function):
```

Comment:
`tag_resource` is *very* new to botocore, we cannot assume everyone has it.

That might not even be a ClientError, it might be an AttributeError.

Can you test with botocore 1.5.39 and see how this behaves. It's fine to fail if someone is passing tags with an old version of botocore, but we should say "Could not set tags - please upgrade to botocore 1.5.40 or higher"

```
    if tags is not None:
        if not hasattr(client, 'tag_resource'):
            module.fail_json(msg="Tags are not supported by your version of botocore. "
                             "Please update to 1.5.40 or newer")
```